### PR TITLE
Rename `minetest.register_async_metatable` to `minetest.register_portable_metatable`

### PR DESCRIPTION
**Pull request by y5nw**
Renames `minetest.register_async_metatable` to `minetest.register_portable_metatable` as noted in https://github.com/minetest/minetest/pull/14369#issuecomment-2106317910

- Goal of the PR
  See above
- How does the PR work?
  This renames a single API function. A deprecation warning is not given since the renamed API function is not (yet) part of a release. An alias with a deprecation warning is kept for `core.register_async_metatable`.
- Does it resolve any reported issue?
  No.
- Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?
  No.
- If not a bug fix, why is this PR needed? What usecases does it solve?
  See linked comment.

## To do

This PR is a Ready for Review.

## How to test

* Check that all tests are passed.
* Check that the documentation is changed accordingly.

### CODE REVIEW

File: builtin/common/metatable.lua [hunk 1]
```diff
@@ -1,6 +1,6 @@
 -- Registered metatables, used by the C++ packer
 local known_metatables = {}
-function core.register_async_metatable(name, mt)
+function core.register_portable_metatable(name, mt)
```
Comment (sfan5) [inline builtin/common/metatable.lua:3]: my only concern: does it make sense to alias this function for existing mods written against 5.9.0-dev?

**Review by sfan5 — APPROVED** [review]
lgtm

File: builtin/common/metatable.lua (hunk 1 above)
Comment (y5nw) [reply builtin/common/metatable.lua:3]: That would make sense.

File: builtin/common/metatable.lua [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 	assert(type(name) == "string", ("attempt to use %s value as metatable name"):format(type(name)))
 	assert(type(mt) == "table", ("attempt to register a %s value as metatable"):format(type(mt)))
 	assert(known_metatables[name] == nil or known_metatables[name] == mt,
 			("attempt to override metatable %s"):format(name))
 	known_metatables[name] = mt
 	known_metatables[mt] = name
 end
+core.register_async_metatable = core.register_portable_metatable
```
Comment (sfan5) [inline builtin/common/metatable.lua:11]: should be a wrapper that calls `core.log("depreated", ...)`

Comment (y5nw) [reply builtin/common/metatable.lua]: Done.

**Review by grorp — APPROVED** [review]
Looks good. I think an alias is unnecessary since the old name only existed in a dev version, but it doesn't hurt either.

**Comment by y5nw** [comment]
I added an alias because I noticed that `minetest.register_async_metatable` is already listed as an API: https://api.minetest.net/minetest-namespace-reference/#async-environment

**Comment by grorp** [comment]
The documentation at api.minetest.net is for the latest dev version. Off-topic: This is confusing for modders and should be changed.

**Pull request merged**